pkg/cli/image/extract/extract.go

```diff
@@ -375,17 +386,33 @@ func (o *ExtractOptions) Run() error {
 					return fmt.Errorf("unable to connect to image repository %s: %v", from.String(), err)
 				}
 
-				srcManifest, location, err := imagemanifest.FirstManifest(ctx, from.Ref, repo, o.FilterOptions.Include)
+				srcManifest, location, err := retrieveSourceManifest(ctx, from, repo, o)
 				if err != nil {
-					if imagemanifest.IsImageForbidden(err) {
-						msg := fmt.Sprintf("image %q does not exist or you don't have permission to access the repository", from)
-						return imagemanifest.NewImageForbidden(msg, err)
-					}
-					if imagemanifest.IsImageNotFound(err) {
-						msg := fmt.Sprintf("image %q not found: %s", from, err.Error())
-						return imagemanifest.NewImageNotFound(msg, err)
+					if strings.Contains(err.Error(), "a manifest image could not be found for this platform") {
```

Comment:
Is `a manifest image could not be found for this platform` an error?. If not, we need to find a matching error to compare with `err`. If yes, `if err == ManifestNotFound` would be better instead calling `Contains` function.

Reply:
I'm not sure exactly what you mean. It seems like an error to me, i.e., I was looking for this os/arch related manifest and it wasn't found. It's not an official error message in the Docker API as far as I know. 

`FirstManifest` returns an error message when there was no matching os/arch manifest [here](https://github.com/openshift/oc/blob/39dfced39bc2e2ce646c85837a5acf1ce081cb50/pkg/cli/image/manifest/manifest.go#L299).

That error message is processed in `IsFilteredAllImagesFromManifestList` [here](https://github.com/bryan-cox/oc/blob/0ea071e40fc4699cf87648a2e8aafc531994838a/pkg/cli/image/extract/extract.go#L571).